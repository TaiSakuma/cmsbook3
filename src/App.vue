<template>
  <v-app>
    <v-navigation-drawer v-model="drawer" app clipped>
      <SideNavi></SideNavi>
    </v-navigation-drawer>
    <v-app-bar app dense clipped-left>
      <v-app-bar-nav-icon @click="drawer = !drawer" />
      <v-toolbar-title>
        <router-link to="/" style="text-decoration: none; color: inherit;">{{ $store.state.title }}</router-link>
      </v-toolbar-title>
      <v-spacer></v-spacer>
      <v-tooltip left open-delay="800">
        <template v-slot:activator="{ on }">
          <v-btn icon @click="$vuetify.theme.dark = !$vuetify.theme.dark" v-on="on">
            <v-icon>mdi-invert-colors</v-icon>
          </v-btn>
        </template>
        <span>Toggle dark mode</span>
      </v-tooltip>
      <template v-slot:extension>
        <TopNavi></TopNavi>
      </template>
    </v-app-bar>
    <v-main>
      <transition name="fade" mode="out-in">
        <router-view :key="$route.path"></router-view>
      </transition>
    </v-main>
  </v-app>
</template>

<script>
import SideNavi from "@/components/SideNavi";
import TopNavi from "@/components/TopNavi";

export default {
  name: "App",
  components: {
    SideNavi,
    TopNavi
  },
  metaInfo() {
    const title= this.$store.state.title;
    return {
      title: title,
      titleTemplate: "%s | " + title
    };
  },
  data: () => ({
    drawer: null
  }),
  mounted() {
    this.$vuetify.theme.dark = localStorage.dark === "true";
    this.$store.dispatch("loadTitle");
    this.$store.dispatch("loadChapters");
  },
  watch: {
    "$vuetify.theme.dark": function(v) {
      localStorage.dark = v;
    }
  }
};
</script>

<style>
.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.2s;
}
.fade-enter,
.fade-leave-active {
  opacity: 0;
}
</style>
