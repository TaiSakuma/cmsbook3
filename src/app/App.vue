<template>
  <v-app>
    <navigation-drawer v-model="drawer"></navigation-drawer>
    <app-bar :order="order">
      <template v-slot:prepend>
        <v-app-bar-nav-icon @click="toggleDrawer" v-if="mobile">
        </v-app-bar-nav-icon>
      </template>
    </app-bar>
    <v-main>
      <router-view v-slot="{ Component }">
        <v-fade-transition>
          <component :key="route.fullPath" :is="Component" />
        </v-fade-transition>
      </router-view>
    </v-main>
  </v-app>
</template>

<script setup lang="ts">
import { ref, computed, watchEffect } from "vue";
import { useDisplay } from "vuetify";
import { useRoute } from "vue-router";

import { useSetTitle } from "./set-title";
import { useDarkMode } from "./dark-mode";

import AppBar from "./AppBar.vue";
import NavigationDrawer from "@/components/NavigationDrawer.vue";

useSetTitle();
useDarkMode();

const { mobile } = useDisplay();

const drawer = ref(true);
watchEffect(() => {
  drawer.value = !mobile.value;
});
function toggleDrawer() {
  drawer.value = !drawer.value;
}

const order = computed(() => (mobile.value ? 0 : -1));

const route = useRoute();
</script>
