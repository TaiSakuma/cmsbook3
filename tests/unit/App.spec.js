import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { mount, shallowMount, createLocalVue } from "@vue/test-utils";

import App from "@/App.vue";

Vue.use(Vuetify);
// Vue.use(VueRouter);

describe("App.vue", () => {
  let localVue;
  let vuetify;
  let router;
  let wrapper;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    router = new VueRouter();
    wrapper = shallowMount(App, {
      localVue,
      vuetify,
      router,
      mocks: {
        $route: {
          path: "/A/b/3.md"
        }
      },
      stubs: ["router-link", "router-view"]
    });
  });

  it("match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
