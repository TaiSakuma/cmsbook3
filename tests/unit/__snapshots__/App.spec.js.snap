// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue match snapshot 1`] = `
<v-app-stub id="app">
  <v-navigation-drawer-stub app="true" mobilebreakpoint="1264" clipped="true" height="100vh" minivariantwidth="56" src="" tag="nav" width="256">
    <sidenavi-stub></sidenavi-stub>
  </v-navigation-drawer-stub>
  <v-app-bar-stub tag="header" dense="true" extensionheight="48" src="" value="true" app="true" clippedleft="true">
    <v-app-bar-nav-icon-stub></v-app-bar-nav-icon-stub>
    <v-toolbar-title-stub>
      <router-link-stub to="/" style="text-decoration: none;">cmsbook</router-link-stub>
    </v-toolbar-title-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-tooltip-stub opendelay="800" closedelay="0" contentclass="" fixed="true" left="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" tag="span"> <span>Toggle dark mode</span></v-tooltip-stub>
  </v-app-bar-stub>
  <v-content-stub tag="main">
    <transition-stub name="fade" mode="out-in">
      <router-view-stub></router-view-stub>
    </transition-stub>
  </v-content-stub>
</v-app-stub>
`;
